src/core/reactor_backend.cc

```diff
@@ -1559,16 +1559,16 @@ class reactor_backend_uring final : public reactor_backend {
     virtual void wait_and_process_events(const sigset_t* active_sigmask) override {
         _smp_wakeup_completion.maybe_rearm(*this);
         _hrtimer_completion.maybe_rearm(*this);
-        ::io_uring_submit(&_uring);
         bool did_work = false;
         did_work |= _preempt_io_context.service_preempting_io();
         did_work |= std::exchange(_did_work_while_getting_sqe, false);
         if (did_work) {
+            ::io_uring_submit(&_uring);
```

Comment:
error handling?

Reply:
ah, there wasn't before.

Reply:
how can you tell what io_uring_submit does? the documentation just says it submits events.

Reply:
> ah, there wasn't before.

will add the error handling in a separated PR.

Reply:
> how can you tell what io_uring_submit does? the documentation just says it submits events.

i don't understand your question. is it related to this change? or do you propose that we should *only* handle the events instead of trying to submit the requests as well. as this function is called when the reactor has no more work to do, and is going to sleep / sleeping.

Reply:
> > ah, there wasn't before.
> 
> will add the error handling in a separated PR.

#1888